Documentation updates

jeffreyhanson opened the pull request:
This PR aims to help improve documentation. Specifically, it adds additional details and examples to the function documentation and adds a vignette (per #12). As discussed over email, I have also updated the DESCRIPTION to list myself as a contributor. To try and make this PR as easy as possible to review, I have made multiple commits related to specific changes. Is this actually helpful? If not, please let me know and I can squash all changes into a single commit.

jeffreyhanson committed:
fix file extension to be consistent

jeffreyhanson committed:
add system file to rbuildignore

jeffreyhanson committed:
remove files accidentally committed by github actions

jeffreyhanson committed:
add github actions files to gitignore

jeffreyhanson committed:
rebuild README.md not README.html

jeffreyhanson committed:
minor README tweaks
- standardize capitalization
- add badge for docs github actions

jeffreyhanson committed:
update DESCRIPTION
- update contributors (per email)
- add knitr for vign

jeffreyhanson committed:
add rmarkdown pkg needed for vigns

jeffreyhanson committed:
ignore temp vign files

jeffreyhanson committed:
remove mistake readme html

jeffreyhanson committed:
add vign, fix #12

jeffreyhanson committed:
update function docs
- add examples
- add detailed descriptions
- add details for cbc_args

jeffreyhanson committed:
manually rebuild docs
- make it easier to review changes

jeffreyhanson commented:
I wasn't sure exactly what information needed to be in the vignette, but hopefully it provides a useful starting point for new users? Let me know if there's additional topics/functionality that should be covered? Also, please feel free to edit/change/update any of the documentation in this PR if anything's not clear (or in case of typos)?

jeffreyhanson commented:
Also, since this PR only covers documentation-related stuff, the Windows CI checks are still failing (per #34).

jeffreyhanson committed:
fix ubuntu github ci
- add missing dep for covr pkg

dirkschumacher commented:
Thank you so much 🙏. I will try to take a deeper look before merging.

jeffreyhanson commented:
Ah - now we've got merge conflict issues due to the automated package website rebuild. Sorry about that - I'll take care of this now.

jeffreyhanson committed:
fix merge conflict

jeffreyhanson committed:
add repo + website to DESCRIPTION

jeffreyhanson commented:
After playing around some more with the rcbc, I realized that the CBC parameters should be in lower case to actually work. Currently, the documentation in this PR incorrectly provides the parameters in upper case. I'll update the documentation to address this mistake.

jeffreyhanson committed:
fix capitalization of params

jeffreyhanson committed:
rebuild site

jeffreyhanson committed:
register native routines

jeffreyhanson commented:
Ah - I'm sorry, I just realized this PR has merge conflict issues. I'll try and address them now.

jeffreyhanson committed:
resolve merge conflict

jeffreyhanson commented:
I'm sorry, this PR has a lot of unnecessary commits. Is that an issue? Please let me know if you would like me to squash all this into a single commit, or open up a new PR (based on a new branch) with just the relevant changes copy-pasted from this branch?

dirkschumacher commented on .github/workflows/check-ubuntu.yaml:
```diff
@@ -61,7 +61,7 @@ jobs:
       - name: Install system dependencies
         run: |
-          sudo apt-get -y install coinor-libcbc-dev coinor-libclp-dev
+          sudo apt-get -y install coinor-libcbc-dev coinor-libclp-dev libcurl4-openssl-dev
```
Why is `libcurl4-openssl-dev` needed?

jeffreyhanson replied:
I'm sorry, that's probably a mistake. I think I included this in the documentation workflow (also based on Ubuntu) to ensure all system dependencies were present for the pkgdown R package -- but this shouldn't be needed here. I think I was sloppy with copy-pasting stuff between the two workflow files. I can delete this if that would be helpful?

dirkschumacher replied:
I think it would be good to delete that

jeffreyhanson replied:
I'll just delete `libcurl4-openssl-dev` now since it's not needed.

jeffreyhanson replied:
> I think it would be good to delete that

Sorry, I didn't see that comment of yours before I posted my previous comment.

dirkschumacher commented on vignettes/rcbc.Rmd:
````diff
@@ -0,0 +1,251 @@
+
+```{r, include = FALSE}
+# load package for generating vignette
+devtools::load_all()
````
Why the call to `devtools` here? Would't that create another dependency?

jeffreyhanson replied:
I've included `devtools::load()` here so that the package vignette is rebuilt using the developmental/latest version of the package. Although one could just install the developmental version of the package locally before trying to rebuild the vignettes, I've found this approach easier to work with. But that's just my opinion - I'm happy to remove this if you want?

Also, technically, devtools isn't detected as a dependency because the vignette has some R code that will stop all code chunks from being evaluated if the vignette code is being run/evaluated as part of R package checks (see https://github.com/jeffreyhanson/rcbc/blob/vign/vignettes/rcbc.Rmd#L15-L23). This is mainly beneficial if (1) the time needed to run the code in the vignette exceeds the limit imposed by CRAN (I explained how this works here in a bit more detail if helpful: https://stackoverflow.com/a/46482580), or (2) you want to include code in vignette that can't run on CRAN (e.g. let's say you wanted to include some code in a vignette that required installation of an R package not available on CRAN, such as gurobi). I don't think (2) is relevant here, but I just wanted to note that for completeness.

I can easily remove this extra stuff if you don't think it's worth it though?

dirkschumacher replied:
Interesting. Does it really take that long to run the vignette? I thought the vignette would always be built with the current package version when it is built (CRAN, pkgdown etc.). But I rarely write vignettes :)

My gut feeling says to not do that unless really needed.

jeffreyhanson replied:
Yeah that's a good point. I rebuilt the vignette locally and recorded that it took only 40 seconds to rebuild. So, it looks like all this extra stuff to handle computationally heavy vignettes isn't needed in this particular case. I'll remove the `devtools::load_all()` and extra stuff so that the vignette follows standard practices.

jeffreyhanson replied:
Ok, I've just removed all the extra stuff so that the vignette follows standard practices. I've also tweaked the .gitignore and documentation workflow to ensure that junk files won't be committed to the repo (i.e `./README.html`, `.github/depends.rds` and `.github/R-version`).

dirkschumacher commented on DESCRIPTION:
```diff
@@ -4,20 +4,36 @@ Title: COIN CBC MILP Solver Bindings
+         role = "ctb"),
+  person(c("Jeffrey", "O"), "Hanson",
+         email="jeffrey.hanson@uqconnect.edu.au",
+         role = c("ctb"),
```
I think the role should be `aut` here if you like.

jeffreyhanson replied:
Yeah that would be great - thanks! I'll update the role to `aut` now.

dirkschumacher commented:
Really great stuff. Thanks for writing that and also adding some literature references.

jeffreyhanson committed:
update Hanson to aut in DESCRIPTION

jeffreyhanson commented:
No worries - I'm happy to help! Thanks so much for making the time to look at this.

jeffreyhanson committed:
remove uneeded dep from ubuntu ci

jeffreyhanson committed:
remove uneeded code from vign

jeffreyhanson committed:
ensure that docs workflow doesnt commit junk, #47

jeffreyhanson commented:
Ah - I've introduced another merge conflict. I'll fix that now.

jeffreyhanson committed:
fix bug in vign

jeffreyhanson committed:
fix merge conflict

jeffreyhanson commented:
Ok - I think I've addressed all the issues with this PR. @dirkschumacher, please let you know if there's any further I changes I can make to improve it?

dirkschumacher commented:
Great thanks. LGTM!

jeffreyhanson commented:
Awesome - thanks! I'll work on updating the README now and adding citation information (e.g. so the users are told to cite the rcbc package and also CBC when using this package).
